internal/services/postgres/postgresql_flexible_server_resource.go

```diff
@@ -278,6 +278,7 @@ func resourcePostgresqlFlexibleServer() *pluginsdk.Resource {
 
 			"public_network_access_enabled": {
 				Type:     pluginsdk.TypeBool,
+				Optional: true,
 				Computed: true,
```

Comment:
this shouldn't be Computed, but defaulted to `true`:

```suggestion
				Default: true,
```

Reply:
@tombuildsstuff , I assume that we can't simply set the default value to "true" based on below reasons.

1. The default value of "public_network_access_enabled" is dynamic. Below is the scenarios. So we can't simply hard-code the default value in the schema. 
```
~> **Note:** When `delegated_subnet_id` and `private_dns_zone_id` are set, the default value of `public_network_access_enabled` is `false`.
~> **Note:** When `delegated_subnet_id` and `private_dns_zone_id` aren't set, the default value of `public_network_access_enabled` is `true`.
```

2. If we set the default value of "public_network_access_enabled" to "true" in the schema, then "public_network_access_enabled" would also be set to "true" in the request payload when it isn't set. At this time, if "delegated_subnet_id" and "private_dns_zone_id" are also set in the request payload, then service API would return below error message.
```
performing Create: unexpected status 400 (400 Bad Request) with error: ConflictingPublicNetworkAccessAndVirtualNetworkConfiguration: Conflicting configuration is detected between Public Network Access and Virtual Network arguments. Public Network Access is not supported along with Virtual Network feature.
```

Reply:
Yes, users would need to set the value to `false` in their configs when they configure either `delegated_subnet_id` or `private_dns_zone_id`, that's expected.

Reply:
Updated